rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,657 @@
+              Represents a partial success response.
+              Server's do not need to implement this.
+    ListNamespacesResponse:
+      type: object
+      properties:
+        namespaces:
+          type: array
+          items:
+            $ref: '#/components/schemas/Namespace'
+    DropTableResponse:
+      type: object
+      properties:
+        dropped:
+          type: boolean
+          description: true if the table was found and removed from the catalog
+        purged:
+          type: boolean
+          description: whether the underlying data was purged or is being purged
+    Namespace:
+      type: array
+      description: Reference to one or more levels of a namespace
+      items:
+        type: string
+      example: [ "accounting", "tax" ]
+    GetNamespaceResponse:
+      type: object
+      properties:
+        namespace:
+          $ref: '#/components/schemas/Namespace'
+        properties:
+          type: object
+          example: { "owner": "Ralph", 'transient_lastDdlTime': '1452120468' }
+    ListTablesResponse:
+      type: object
+      properties:
+        identifiers:
+          type: array
+          items:
+            $ref: '#/components/schemas/TableIdentifier'
+    IcebergConfiguration:
+      type: object
+      description: "Server-provided configuration for the catalog."
+      properties:
+        rootPath:
+          minLength: 1
+          type: string
+          default: /
+          nullable: false
+          description:
+            Root path to be used for all other requests. Server-side implementations
+            are free to use another choice for root path, but must conform to the
+            specification otherwise in order to interoperate with the URI generation
+            that the REST catalog client will do.
+        catalogProperties:
+          type: object
+          nullable: true
+          description:
+            An optional field for storing catalog configuration properties that
+            are stored server side. This could be beneifical to an administrator,
+            for example to enforce that a given LocationProvider is used or to
+            enforce that a certain FileIO implementation is used.
+
+    TableAlreadyExistsError:
+      allOf:
+        - $ref: '#/components/schemas/ResponseErrorObject'
+      description: the given table identifier already exists and cannot be created / renamed to
+
+  #############################
+  # Reusable Response Objects #
+  #############################
+  responses:
+
+    IcebergResponseObject:
+      description: JSON wrapper for all response bodies, with a data object and / or an error object
+      content:
+        application/json:
+          schema:
+            anyOf:
+              - $ref: '#/components/schemas/ResponseDataObject'
+              - $ref: '#/components/schemas/ResponseErrorObject'
+          example: { "data": { "namespaces": [ [ "ns1", "foo_db" ], [ "ns1", "bar_db" ] ] }, "error": { } }
```

Comment:
should it be `anyOf` or `oneOf`? Wouldn't that make it even more confusing if the response has both a data and an error field?

Reply:
I was thinking in the case of partial success and partial error (such as in a microbatch process of statements), which is where I've used that. Though that can usually be defined in the response body inside just the data-intended section.

So mostly likely it should be `oneOf`.

Reply:
Feels like a `oneOf` would effectively make `IcebergResponseObject` superfluous and operations can specify the "right" type for each response without an intermediate `ResponseDataObject` or `ResponseErrorObject`?

Reply:
I think if ResponseDataObject/ResponseErrorObject just contain a generic JSON doc/object, clients have to manually construct the right object type.

Reply:
I am open to better ways to format the OpenAPI doc to express this. We have agreed on using `error` at least, but if there's a better way in OpenAPI to define it, I'd be happy to hear it or possibly a link to a description or example would help. 😄